src/templater.rs

```diff
@@ -239,7 +239,7 @@ impl TemplateProperty<Commit, String> for BranchProperty<'_> {
             if let Some(local_target) = local_target {
                 if local_target.has_add(context.id()) {
                     if local_target.is_conflict() {
-                        names.push(format!("{branch_name}?"));
+                        names.push(format!("{branch_name}??"));
```

Comment:
Ah, I was going to ask for this change for consistency, and to make it easier to spot in the output. Thanks :)

Reply:
Good to hear!

Do you have any thoughts about making it red? I talked about it in the commit description, it requires similar work to what's needed to make id prefixes have a different color (or be underlined).

Reply:
Good question.

I wonder if we should instead create a type that represents a structured string so we can have `commit_id` return something like `<commit_id><unique>ab</unique>c123</commit_id>` and the branches can be `<branch><conflicted>main</conflicted></branch>` or something like that. Those labels there would not actually be in the string, of course; we would have some type representing that structure.

The disadvantage is that it would make not be possible to do `commit_id.unique() commit_id.remainder()` like it would if we instead made the parts accessible as methods. So I'm not sure which is better. If we returned a single structured string, it would also be easier to make it avoid calculating the unique prefix twice (once for the prefix and once for the remainder).

What do you think?

Reply:
Ideally, I think a custom type should be returned that includes all the information needed to print the commit (or branch) in whichever format. It could have a reference to the commit (or branch) object and all the formatting information (e.g. which branches are conflicted, how many letters in the unique prefix for an id, etc.)

If that turns out to be difficult to create, we could go in the structured string direction. It's currently harder than I would like to modify the Property enum, but adding two things to it should be doable, or it could perhaps be refactored somehow.

Reply:
I'll resolve this for the purposes of this PR, but we should certainly continue discussing it.

Reply:
We could create a type like that, but we probably don't want to populate it with data eagerly, because some pieces can be expensive to calculate and we don't want a simple `jj log -T 'commit_id "\n"'` to have to figure out which commits have conflicts etc, unless possibly if we make all the information we might want to print indexed so it's fast enough to produce that struct with all the values in it.

Reply:
For the unique ids, I wanted to store the unique prefixes in ReadonlyRepo, lazily generated. It has to read the whole repository to generate them anyway. Later, I wanted the divergence indicator and lookup of commits by prefix to use the same structure.

The other thing I was thinking about is whether something like `CommitIdKeyword` could be that formatting structure, or whether to make a new one. The later is easier to do, but less elegant.